thrift/server.cc

```diff
@@ -148,18 +149,17 @@ future<>
 thrift_server::connection::process_one_request() {
     _input->resetBuffer();
     _output->resetBuffer();
-    return read().then([this] {
-        ++_server._requests_served;
-        auto ret = _processor_promise.get_future();
-        // adapt from "continuation object style" to future/promise
-        auto complete = [this] (bool success) mutable {
-            // FIXME: look at success?
-            write().forward_to(std::move(_processor_promise));
-            _processor_promise = promise<>();
-        };
-        _processor->process(complete, _in_proto, _out_proto);
-        return ret;
-    });
+    co_await read();
+    ++_server._requests_served;
+    auto ret = _processor_promise.get_future();
+    // adapt from "continuation object style" to future/promise
+    auto complete = [this] (bool success) mutable {
+        // FIXME: look at success?
+        write().forward_to(std::move(_processor_promise));
+        _processor_promise = promise<>();
+    };
+    _processor->process(complete, _in_proto, _out_proto);
+    co_return co_await ret;
```

Comment:
co_return co_await is slightly less efficient than what we had previously (it needs one more continuation), but I am guessing this deliberate and you'll need it later to keep the permit alive.
In any case, I guess the cost of the extra continuation is tiny, and perhaps even offset by other differences beween the coroutine and future version - so I think this change is fine.